File: Median Of Two Sorted Arrays/Median Of Two Sorted Arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(int A[], int m, int B[], int n) {
        int len = m + n; 
        if (len % 2 == 0) 
            return ((double)findKthSortedArrays(A, 0, m, B, 0, n, len/2) + (double)findKthSortedArrays(A, 0, m, B, 0, n, len/2 + 1)) / 2;
        else 
            return (double)findKthSortedArrays(A, 0, m, B, 0, n, len/2 + 1);
    }
    double findKthSortedArrays(int A[], int startOfA, int m, int B[], int startOfB, int n, int k) {
        if (startOfA >= m) return B[startOfB + k - 1];
        if (startOfB >= n) return A[startOfA + k - 1];
        
        if (k == 1) return min(A[startOfA], B[startOfB]);
        
        int keyA;
        int keyB;
        if (startOfA + k/2 -1 >= m) keyA = INT_MAX;
        else keyA = A[startOfA + k/2 -1];
        if (startOfB + k/2 -1 >= n) keyB = INT_MAX;
        else keyB = B[startOfB + k/2 -1];
        
        if (keyA < keyB) return findKthSortedArrays(A, startOfA + k/2, m, B, startOfB, n, k - k/2 );
        else return findKthSortedArrays(A, startOfA, m, B, startOfB + k/2, n, k - k/2);
    }
};

File: Median Of Two Sorted Arrays/Median Of Two Sorted Arrays.java
public class Solution {
    public double findMedianSortedArrays(int A[], int B[]) {
        int len = A.length + B.length; 
        if (len % 2 == 0) 
            return ((double)findKthSortedArrays(A, 0, B, 0, len/2) + (double)findKthSortedArrays(A, 0, B, 0, len/2 + 1)) / 2;
        else 
            return (double)findKthSortedArrays(A, 0, B, 0, len/2 + 1);
    }
    private double findKthSortedArrays(int A[], int startOfA, int B[], int startOfB, int k) {
        if (startOfA >= A.length) return B[startOfB + k - 1];
        if (startOfB >= B.length) return A[startOfA + k - 1];
        
        if (k == 1) return Math.min(A[startOfA], B[startOfB]);
        
        int keyA;
        int keyB;
        if (startOfA + k/2 -1 >= A.length) keyA = Integer.MAX_VALUE;
        else keyA = A[startOfA + k/2 -1];
        if (startOfB + k/2 -1 >= B.length) keyB = Integer.MAX_VALUE;
        else keyB = B[startOfB + k/2 -1];
        
        if (keyA < keyB) return findKthSortedArrays(A, startOfA + k/2, B, startOfB, k - k/2 );
        else return findKthSortedArrays(A, startOfA, B, startOfB + k/2, k - k/2);
    }
}